src/exercises/concurrency/dining-philosophers-async.rs

```diff
@@ -72,12 +72,15 @@ async fn main() {
         let mut philosophers = vec![];
         let (tx, rx) = mpsc::channel(10);
         for (i, name) in PHILOSOPHERS.iter().enumerate() {
-            let left_fork = forks[i].clone();
-            let right_fork = forks[(i + 1) % PHILOSOPHERS.len()].clone();
+            let left_fork = Arc::clone(&forks[i]);
+            let right_fork = Arc::clone(&forks[(i + 1) % PHILOSOPHERS.len()]);
+            if i % 2 == 1 {
```

Comment:
I'm not actually sure why this is done for every _second_ philosopher? Why not just swap it for a single philosopher like the synchronous version?

Reply:
Good point. That should work. Do you want me to fix it, or will you prefer to fix it in this PR?

Reply:
> Do you want me to fix it, or will you prefer to fix it in this PR?

Cool, I think this should do it:

```suggestion
            // To avoid a deadlock, we have to break the symmetry
            // somewhere. This will swap the forks without deinitializing
            // either of them.
            if i  == 0 {
```

Reply:
Nice. Thanks.